release/context.go

```diff
@@ -112,7 +112,11 @@ func (rc *ReleaseContext) SelectServices(results flux.ReleaseResult, logStatus s
 	var updates []*ServiceUpdate
 	for _, s := range services {
 		logStatus("Found service %s", s.ID)
-		update := definedMap[s.ID]
+		update, ok := definedMap[s.ID]
```

Comment:
We take the ids from the defined services, then look up only those in the running system, so the running services we found *should* only ever be a subset. It was panicking in a test with mocks returning slightly odd results. Figured no harm in leaving in the check.